R/import_config.R

```diff
@@ -88,31 +115,30 @@ set_missing_values <- function(info, field_name) {
   info
 }
 
-#' Read file fields of the dettl config yaml file.
+#' Read function fields of the dettl config yaml file.
 #'
 #' Reads the config file, and sources any file references.
 #'
-#' @param fields The name of the field.
+#' @param fields List of fields with functions and whther they have to exist.
 #' @param config The config being read.
 #' @param env Environment containing functions loaded from sources.
 #' @keywords internal
 #'
 read_function_fields <- function(fields, config, env) {
-  for (field in fields) {
-    if (field != "load" || !config$load$automatic) {
-      assert_func_exists(config[[field]]$func, env)
-      config[[field]]$func <- get0(config[[field]]$func,
-                                   envir = env,
-                                   mode = "function", inherits = FALSE
-      )
-    }
-    if (field == "load") {
-      assert_func_exists(config[[field]]$verification_queries, env)
-      config[[field]]$verification_queries <-
-        get0(config[[field]]$verification_queries,
-             envir = env,
-             mode = "function", inherits = FALSE
-        )
+  for (field in names(fields)) {
+    for (property in fields[[field]]) {
+      func_name <- config[[field]][[property$func]]
+      if (is.null(func_name) && property$must_exist) {
```

Comment:
Almost seems like it's a required/optional/allowed set of cases here that we need